dpnp/dpnp_utils/dpnp_utils_linearalgebra.py

```diff
@@ -1894,108 +2024,99 @@ def dpnp_matmul(
             elif len(axes_res) == 1:
                 out = dpnp.moveaxis(out, axes_res, (-1,))
```

Comment:
I modified the test to add `out` keyword. However, the code itself does not need any modification since `len(axes_res) == 0` happens when out is 0D array and there is no need for moving axis.